compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/ConstantValueResolver.java

```diff
@@ -86,8 +86,10 @@ public void visit(BLangConstant constant) {
         BConstantSymbol tempCurrentConstSymbol = this.currentConstSymbol;
         this.currentConstSymbol = constant.symbol;
         this.currentConstSymbol.value = visitExpr(constant.expr);
-        unresolvedConstants.remove(this.currentConstSymbol);
-        this.currentConstSymbol = tempCurrentConstSymbol;
+        if (this.currentConstSymbol.value != null) {
+            unresolvedConstants.remove(this.currentConstSymbol);
```

Comment:
With this approach wouldn't we try to resolve the constant every time it is referenced even though we know it is an error. Wondering if we should fix the resolving logic instead and consider the key not being present + value being null to indicate an error.

Reply:
Yes I had done a mistake. We have to indicate an error for such scenarios.